_examples/postcard/storage/mapping_postgres_eventstore.go

```diff
@@ -86,25 +88,17 @@ type Created struct {
 	ID string `json:"id"`
 }
 
-type CreatedMapper struct{}
-
-func (CreatedMapper) SupportedEvent() stream.Event[postcard.Postcard] {
+func (e Created) SupportedEvent() stream.Event[postcard.Postcard] {
```

Comment:
I see the simplification you're aiming at here, and I also thought about it. But it also feels weird that the type is used for both instantiation and returning the `SupportedEvent`.

What do you think of passing a map instead and getting rid of this method?

```go
map[stream.Event[postcard.Postcard]]transport.Event[postcard.Postcard]{
    postcard.Created{}: Created{},
}
```

The definition looks terrible, though. So maybe a slice of structs with two values? Or just use `EventName` as the key.

Reply:
Let me try, sure!

Reply:
https://github.com/ThreeDotsLabs/esja/pull/18/commits/fedd53c8f895b950525413b835cfceadfdcbd631 Check this out:)

Reply:
nice!